libglusterfs/src/monitoring.c

```diff
@@ -90,46 +92,46 @@ dump_latency_and_count(xlator_t *xl, int fd)
     if ((xl != xl->ctx->primary) && (xl->ctx->active != xl->graph))
         return;
 
-    count = GF_ATOMIC_GET(xl->stats.total.count);
-    dprintf(fd, "%s.total.fop-count %" PRIu64 "\n", xl->name, count);
-
-    count = GF_ATOMIC_GET(xl->stats.interval.count);
-    dprintf(fd, "%s.interval.fop-count %" PRIu64 "\n", xl->name, count);
-    GF_ATOMIC_INIT(xl->stats.interval.count, 0);
-
     for (index = 0; index < GF_FOP_MAXVALUE; index++) {
-        fop = GF_ATOMIC_GET(xl->stats.total.metrics[index].fop);
+        fop = GF_ATOMIC_GET(xl->stats[index].total_fop);
         if (fop) {
             dprintf(fd, "%s.total.%s.count %" PRIu64 "\n", xl->name,
                     gf_fop_list[index], fop);
+            total_fop_count += fop;
         }
-        fop = GF_ATOMIC_GET(xl->stats.interval.metrics[index].fop);
+        GF_ATOMIC_SWAP(xl->stats[index].interval_fop, fop);
         if (fop) {
             dprintf(fd, "%s.interval.%s.count %" PRIu64 "\n", xl->name,
                     gf_fop_list[index], fop);
+            interval_fop_count += fop;
+            fop = 0;
```

Comment:
Not needed.

Reply:
interval_fop_count is required because we are printing this.

Reply:
The comment was for line 107.